lib/src/Base/Optim/Cobyla.cxx

```diff
@@ -149,10 +150,11 @@ void Cobyla::run()
     result_.store(inP, Point(1, outP[0]), absoluteError, relativeError, residualError, constraintError);
   }
 
-  result_.setOptimalPoint(x);
-  const UnsignedInteger index = evaluationInputHistory_.find(x);
-  Scalar bestValue = evaluationOutputHistory_(index, 0);
-  result_.setOptimalValue(Point(1, bestValue));
+  // x might not be exactly the best point evaluated, so find the nearest
+  const UnsignedInteger optimalIndex = NaiveNearestNeighbour(evaluationInputHistory_).query(x);
```

Comment:
Why NaiveNearestNeighbour and not NearestNeighbourAlgorithm?

Reply:
Ok, I see it instanciates KDTree which is more efficient.